### PR TITLE
Feature/tofs lookup files in tpldir hierarchy

### PR DESCRIPTION
**Pull request by baby-gnu**
It will ease the use of sub-component with dedicated `files/` directory like in `systemd-formula`.

### CODE REVIEW

**Comment by baby-gnu** [comment]
Testing in my personnal test environnmnt, before this pull request, the `template.config.file` was:

```
# -*- coding: utf-8 -*-
# vim: ft=sls

include:
  - template.package.install

template-config-file-file-managed:
  file.managed:
    - name: /etc/template-formula.conf
    - source:
      - salt://template/files/any/path/can/be/used/here/example.tmpl
      - salt://template/files/any/path/can/be/used/here/example.tmpl.jinja
      - salt://template/files/ca222e29ff6d/example.tmpl
      - salt://template/files/ca222e29ff6d/example.tmpl.jinja
      - salt://template/files/Debian-9/example.tmpl
      - salt://template/files/Debian-9/example.tmpl.jinja
      - salt://template/files/Debian/example.tmpl
      - salt://template/files/Debian/example.tmpl.jinja
      - salt://template/files/Debian/example.tmpl
      - salt://template/files/Debian/example.tmpl.jinja
      - salt://template/files/default/example.tmpl
      - salt://template/files/default/example.tmpl.jinja
    - mode: 644
    - user: root
    - group: root
    - template: jinja
    - require:
      - sls: template.package.install
```

With this pull request it becomes:

```
# -*- coding: utf-8 -*-
# vim: ft=sls

include:
  - template.package.install

template-config-file-file-managed:
  file.managed:
    - name: /etc/template-formula.conf
    - source:
      - salt://template/files/any/path/can/be/used/here/example.tmpl
      - salt://template/files/any/path/can/be/used/here/example.tmpl.jinja
      - salt://template/files/ca222e29ff6d/example.tmpl
      - salt://template/files/ca222e29ff6d/example.tmpl.jinja
      - salt://template/files/Debian-9/example.tmpl
      - salt://template/files/Debian-9/example.tmpl.jinja
      - salt://template/files/Debian/example.tmpl
      - salt://template/files/Debian/example.tmpl.jinja
      - salt://template/files/Debian/example.tmpl
      - salt://template/files/Debian/example.tmpl.jinja
      - salt://template/files/default/example.tmpl
      - salt://template/files/default/example.tmpl.jinja
      - salt://template/config/files/any/path/can/be/used/here/example.tmpl
      - salt://template/config/files/any/path/can/be/used/here/example.tmpl.jinja
      - salt://template/config/files/ca222e29ff6d/example.tmpl
      - salt://template/config/files/ca222e29ff6d/example.tmpl.jinja
      - salt://template/config/files/Debian-9/example.tmpl
      - salt://template/config/files/Debian-9/example.tmpl.jinja
      - salt://template/config/files/Debian/example.tmpl
      - salt://template/config/files/Debian/example.tmpl.jinja
      - salt://template/config/files/Debian/example.tmpl
      - salt://template/config/files/Debian/example.tmpl.jinja
      - salt://template/config/files/default/example.tmpl
      - salt://template/config/files/default/example.tmpl.jinja
    - mode: 644
    - user: root
    - group: root
    - template: jinja
    - require:
      - sls: template.package.install
```

Which add a lookup in the `template/config/files/` directory:

``` diff
--- before	2019-03-26 15:10:31.000896539 +0100
+++ after	2019-03-26 15:10:24.816864044 +0100
@@ -20,6 +20,18 @@
       - salt://template/files/Debian/example.tmpl.jinja
       - salt://template/files/default/example.tmpl
       - salt://template/files/default/example.tmpl.jinja
+      - salt://template/config/files/any/path/can/be/used/here/example.tmpl
+      - salt://template/config/files/any/path/can/be/used/here/example.tmpl.jinja
+      - salt://template/config/files/ca222e29ff6d/example.tmpl
+      - salt://template/config/files/ca222e29ff6d/example.tmpl.jinja
+      - salt://template/config/files/Debian-9/example.tmpl
+      - salt://template/config/files/Debian-9/example.tmpl.jinja
+      - salt://template/config/files/Debian/example.tmpl
+      - salt://template/config/files/Debian/example.tmpl.jinja
+      - salt://template/config/files/Debian/example.tmpl
+      - salt://template/config/files/Debian/example.tmpl.jinja
+      - salt://template/config/files/default/example.tmpl
+      - salt://template/config/files/default/example.tmpl.jinja
     - mode: 644
     - user: root
     - group: root
```

and the `template.subcomponent.config` is:

```
# -*- coding: utf-8 -*-
# vim: ft=sls

template-subcomponent-config-file-file-managed:
  file.managed:
    - name: /etc/template-subcomponent-formula.conf
    - source:
      - salt://template/files/any/path/can/be/used/here/subcomponent-example.tmpl
      - salt://template/files/any/path/can/be/used/here/subcomponent-example.tmpl.jinja
      - salt://template/files/ca222e29ff6d/subcomponent-example.tmpl
      - salt://template/files/ca222e29ff6d/subcomponent-example.tmpl.jinja
      - salt://template/files/Debian-9/subcomponent-example.tmpl
      - salt://template/files/Debian-9/subcomponent-example.tmpl.jinja
      - salt://template/files/Debian/subcomponent-example.tmpl
      - salt://template/files/Debian/subcomponent-example.tmpl.jinja
      - salt://template/files/Debian/subcomponent-example.tmpl
      - salt://template/files/Debian/subcomponent-example.tmpl.jinja
      - salt://template/files/default/subcomponent-example.tmpl
      - salt://template/files/default/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/any/path/can/be/used/here/subcomponent-example.tmpl
      - salt://template/subcomponent/files/any/path/can/be/used/here/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/ca222e29ff6d/subcomponent-example.tmpl
      - salt://template/subcomponent/files/ca222e29ff6d/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/Debian-9/subcomponent-example.tmpl
      - salt://template/subcomponent/files/Debian-9/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/Debian/subcomponent-example.tmpl
      - salt://template/subcomponent/files/Debian/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/Debian/subcomponent-example.tmpl
      - salt://template/subcomponent/files/Debian/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/default/subcomponent-example.tmpl
      - salt://template/subcomponent/files/default/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/any/path/can/be/used/here/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/any/path/can/be/used/here/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/ca222e29ff6d/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/ca222e29ff6d/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/Debian-9/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/Debian-9/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/Debian/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/Debian/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/Debian/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/Debian/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/default/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/default/subcomponent-example.tmpl.jinja
    - mode: 644
    - user: root
    - group: root
    - template: jinja
    - require:
      - sls: template.package.install
```

**Comment by baby-gnu** [comment]
And the `template.subcomponent.config.clean` is now an example of how to reload a service when a configuration file change.

**Comment by myii** [comment]
@baby-gnu I thought I had seen the last of TOFS!  This is only a very brief review after the first look because I don't have much time at the moment.  Two things to share for now.

#### Concern about complexity

I understand what the aim is here, to be able to have `files/` sub-directories at any level of the formula hierarchy.  For me, having got involved with the TOFS implementation, it makes sense.  I'd like to find out how others feel about adopting this, since it may feel like it is leading to unnecessary levels of complexity, with an immensely large number of places that a template could be.  That could become a debugging nightmare, especially if formula writers forget to use `{{ source }}` in their templates, or can't use that because they are using another templating engine.  Are formulas supposed to reach this level of configurability?  Or can this be restricted to existing TOFS implementations like the `systemd-formula`?

#### Prioritisation in `source`

Secondly, if this is to go forwards, the logical ordering of `source` should reflect the specific level of the directory in the hierarchy.  So instead of:

```sls
    - source: 
      - salt://template/files/...
      - salt://template/subcomponent/files/...
      - salt://template/subcomponent/config/files/...
```

Use the opposite ordering:

```sls
    - source: 
      - salt://template/subcomponent/config/files/...
      - salt://template/subcomponent/files/...
      - salt://template/files/...
```

**Comment by daks** [comment]
About complexity: TOFS pattern already had a lot of complexity: jinja imports everywhere, every `source` in a `file.managed` is now several lines long, and it seems that we are pushing a lot of code for a lot more flexibility but... I wonder if this flexibility (hence complexity) is already used.

I thought that I understood what TOFS pattern was for: let a formula user override its templates beyond what formula creators have already thought. OK, that seems fair, but that also means it's concerns a minority of users. If really it's the case, I don't get why we are still pushing this pattern further on, by complexifying it.
_But I may have missed the real objective of TOFS..._

Another concern: I think every source costs an I/O on salt master _(which tries to read the file if it exists while it does not find one that exists)_. Doubling them also doubles the cost.

**Comment by baby-gnu** [comment]
@myii: I completely understand your concern about complexity, the `libtofs.jinja` reach 50 lines of jinja (without comments) and as @daks said each new `source` add to the cost if it does not provide the file.

I agree to close this PR and keep the `v1_path_prefix` for formula needing it.

It looks like some people, like @daks, wonder if `libtofs.jinja` is that useful for it's complexity and I think it can be a good point between:

* this formula can be used as-is just by configuring some pillar
* I need to rewrite the formula since it's useless to my use case.

It permits to stay easy on the jinja in the configuration file templates, replacing a lot of program logic by a dedicated file at the proper place in the `files/` hierarchy.

Thanks for the points.

**Comment by myii** [comment]
@baby-gnu Thanks for your critical assessment; that's not easy to do when you've put serious effort into a PR.  I appreciate the work you put in and I am very happy that your implementation still lives on in this PR's diff, if we ever need to reference it.  I can still imagine there will be situations where this comes in useful.

There is no question in my mind that TOFS is useful.  As you've said, since it has sensible defaults, it is very simple to make use of and doesn't cause many entries under `source`.

**Comment by noelmcloughlin** [comment]
Hi @myii 
This should be revisited. The intention to me is that any formula can have a sub-component. This is common pattern in software and systems design. This PR would show how to isolate a subcomponent with its only child TOFS logic.  If the TOFS pattern is applied consistently to parent and subparent components then it's a good pattern as long as parent TOFS configuration does not leak into subcomponent then complexity is not worse.  

But I think the PR implementation is wrong - parent TOFS config should not leak into subcomponent so this is more correct.
```
template-subcomponent-config-file-file-managed:
  file.managed:
    - name: /etc/template-subcomponent-formula.conf
    - source: 
      - salt://template/subcomponent/files/any/path/can/be/used/here/subcomponent-example.tmpl
      - salt://template/subcomponent/files/any/path/can/be/used/here/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/ca222e29ff6d/subcomponent-example.tmpl
      - salt://template/subcomponent/files/ca222e29ff6d/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/Debian-9/subcomponent-example.tmpl
      - salt://template/subcomponent/files/Debian-9/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/Debian/subcomponent-example.tmpl
      - salt://template/subcomponent/files/Debian/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/Debian/subcomponent-example.tmpl
      - salt://template/subcomponent/files/Debian/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/files/default/subcomponent-example.tmpl
      - salt://template/subcomponent/files/default/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/any/path/can/be/used/here/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/any/path/can/be/used/here/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/ca222e29ff6d/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/ca222e29ff6d/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/Debian-9/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/Debian-9/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/Debian/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/Debian/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/Debian/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/Debian/subcomponent-example.tmpl.jinja
      - salt://template/subcomponent/config/files/default/subcomponent-example.tmpl
      - salt://template/subcomponent/config/files/default/subcomponent-example.tmpl.jinja
`

**Comment by myii** [comment]
@noelmcloughlin Thanks for the feedback.  Having contributed to the implementation, everything makes sense and seems like natural progression.  But there have been a number of queries about adopting TOFS and issues with the Jinja.  It's not something I wanted @baby-gnu to close but for the discussion to be opened up, to solicit feedback from others who would use TOFS, without necessarily understanding its implementation.  I'm still very happy to have this discussion and to ask @baby-gnu to reconsider if we agree to adopt this functionality.

**Comment by noelmcloughlin** [comment]
Thanks for being open. I am at a disadvantage having not used TOFS yet. I like the idea of template formula demonstrating a subcomponent, I understood TOFS advantages seem to outweigh the disadvantages - perhaps the complexity of TOFS could be hidden by jina2 macros.  

I would also like to understand how template formula could demonstrate use of two different text and data renderers such as yaml/json and jinja/mako. If we explore these ideas and this PR is certainly contributing to that exploration, then we can learn things. I'm interested in what drove @baby-gnu to raise PR for this innovation - But I would not want to belittle any negative opinions on TOFS.

**Comment by baby-gnu** [comment]
Mouhahaha.

I was resigned to abandon this PR and someone is interested by it ;-)

I reopen it and update the pattern to lookup from `tpldir` to `tplroot`, as explained at the end of `docs/TOFS_pattern.rst`

**Comment by myii** [comment]
@baby-gnu Welcome back!  Let's try to get some conversations going on about this, to work out what is the best way to introduce this.  You have to watch out: @noelmcloughlin wants you to "improve" this further by introducing sub-directories for `jinja/`, `mako/`, etc. instead of only `files/`!!

CC: @aboe76 @javierbertoli @daks: don't be scared off by the massive `source` list showed in the comment at the top -- that's only when someone is using "maximum" configurability.

@baby-gnu One idea: maybe we can use this in a sample project or one of @noelmcloughlin's formulas, so that we can "prove" its worth?  I understand how it works because of my familiarity with `libtofs.jinja` but it is probably very difficult to envisage from just reading the code.

**Comment by baby-gnu** [comment]
Hello.

@myii and @noelmcloughlin I'm not convinced about the per template technologie directory (`jinja/`, `mako/`, etc.).

The `libtofs.jinja` template is for use by SLS writers, a user can't choose to change the templating engine, it's hardcoded in the SLS, as in the `template.config.file`:

```
template-config-file-file-managed:
  file.managed:
    - name: {{ template.config }}
    - source: {{ files_switch(['example.tmpl', 'example.tmpl.jinja'],
                              lookup='template-config-file-file-managed'
                 )
              }}
    - mode: 644
    - user: root
    - group: root
    - template: jinja
    - require:
      - sls: {{ sls_package_install }}
```

If the user wants to use another templating engine, she needs to override the SLS to provide it's own version.

Personnaly I always prefer to put templates in the same directory and use the filename extension to make the difference between a plain file and and template, like the `example.tmpl` and `example.tmpl.jinja`.

Or maybe you found a way to set the `template:` key based on the directory name?

Maybe this particular topic can be discussed in another issue since I think it's not completely related to lookup in hierarchy?

Regards.

**Comment by noelmcloughlin** [comment]
I'm not convinced either. But it's good architecture to consider how to support a feature which is technically support by the underlying infrastructure platform.  Switching template engine via pillar value is straight forward but what would directory structure, and template-formula, look like?  I think  this PR is wrong place to consider what implementation for BYO-Renderer would look like!

Anyway I was interested in this PR because it is demonstrating how an important use case could work. The community may need time to absorb the PR so I think approving/merging/closing could wait a little until community members get time to try TOFS and then give more reasonable feedback?

**Comment by baby-gnu** [comment]
> @noelmcloughlin But I think the PR implementation is wrong - parent TOFS config should not leak into subcomponent so this is more correct [...]

I personally don't know how to make the difference between `template.config.file` and `template.foo.config.file` where only `foo` is a sub-component.

Do you think that the probability of:

* the sub-component `files/` does not provide the wanted template file
* the top level `files/` has a template file with the same name as the one wanted by the sub-component

is enough to search for a mechanism (probably only manual configuration like the one used in `systemd-formula`) to solve this issue?


> CC: @aboe76 @javierbertoli @daks: don't be scared off by the massive `source` list showed in the comment at the top -- that's only when someone is using "maximum" configurability.

By default, in the version in the `master` branch, the `files_switch` macro will produce:

```
- salt://template/files/<minion_id>/
- salt://template/files/<os>/
- salt://template/files/default/
```

This PR will multiply this by the level in the hierarchy under the formula directory:

* `template.config` → 3 lookups max
* `template.config.file` → 6 lookups max
* `template.subcomponent.config.file` → 9 lookups max

**Comment by baby-gnu** [comment]
Hello.

> @noelmcloughlin  Switching template engine via pillar value is straight forward but what would directory structure, and template-formula, look like?

And it make it more complex for the user:

* she provides a `mako` template for a subset of minions
* she relies on the formula default template for everything else

Not everything is solvable by this pattern. With it, it's a little bit easier to provide specific templates for specific minions without the need to override the SLS.

> I think this PR is wrong place to consider what implementation for BYO-Renderer would look like!

I agree, maybe you could open a new issue to follow the ideas?

Regards

**Comment by noelmcloughlin** [comment]
Can be new issue - I have no pressing use case for this - its just good architectural design principle to make sure the implementation supports 80% of use case but anticipates a need to go further. I didn't raise question of multiple renderers in this ticket, that was comment by @myii. We can skip this.

**Comment by myii** [comment]
@baby-gnu @noelmcloughlin Thanks for continuing this conversation.  @noelmcloughlin Do you have any repos with sub-components, but still based upon the single template directory such as `files/`?  I think it would be immensely useful to get this PR out there into a real environment and then bring it back here.  That's how TOFS got incremental improvements after all: real formulas => `template-formula`.

**Comment by noelmcloughlin** [comment]
One example is [opensds formula files/ dir](https://github.com/saltstack-formulas/opensds-formula/tree/master/opensds/files) which many subcomponents using a single `files` top-level directory. Some subcomponents have their own `defaults.yaml`.  The map.jinja for this type of formula gets interesting but work well.

**Comment by myii** [comment]
@noelmcloughlin Does it have any sub-components with a dedicated `files/` sub-directory?  A repo containing that would be a good place to implement these changes.

**Comment by noelmcloughlin** [comment]
> @noelmcloughlin Does it have any sub-components with a dedicated `files/` sub-directory? A repo containing that would be a good place to implement these changes.

No. I cheated by creating a  [config subcomponent](https://github.com/saltstack-formulas/opensds-formula/tree/master/opensds/config):

**Comment by noelmcloughlin** [comment]
> I personally don't know how to make the difference between `template.config.file` and `template.foo.config.file` where only `foo` is a sub-component.
> 
> Do you think that the probability of:
> 
> * the sub-component `files/` does not provide the wanted template file
> * the top level `files/` has a template file with the same name as the one wanted by the sub-component
> 
> is enough to search for a mechanism (probably only manual configuration like the one used in `systemd-formula`) to solve this issue?


Guys @baby-gnu  @myii 
I think I missed this question.
For the record I'll document some examples of component/sub-component use cases.
I'm not sure this helps but 

** TL;DR **
In real life 
... a top-level `files/` directory is unnecessary for components/sub-components config.
The drivers are ...
... useful for macros (i.e. generate json config file, etc) - A valid driver.
... DRY-approach (i.e. ensuring file.directory gets run regardless of state) - questionable driver.

I'm unsure if this PR is needed or not.


** MY EXPERIENCE OF COMPONENT/SUB-COMPONENTS **

**Example 1: Multiple Components, no top-level `template.config.file`.**

Each component wanting a config has local config.file. No sub-components exist.
The meta-state `init.sls` in top-level directory is strictly not a component in this case.
 
(1A) This is the (*wanted*) MongoDB formula structure discussed in some PRs.
`mongodb/mongos/config/init.sls`
`mongodb/mongod/`
`mongodb/bic/config/init.sls`
`mongodb/robo3t/`
`mongodb/compass/`

(1B) Current iscsi formula.

This formula has three components each with top-level config. No sub-components.
(Note: the configuration state should ideally be separated from init state for granularity).
```
iscsi/
iscsi/initiator/install.sls     #< includes configuration states
iscsi/target/install.sls        #< ditto
iscsi/isns/install.sls           #< ditto
```

**Example 2:  Multiple Components with top-level `template.config.file`.**

This is driven by implementation choice (overzealous DRY approach) not need.

(2A)  I think this pattern is not good - DRY vs confusing.
 
`mongodb/config/init.sls`     #<----- centralize duplicated SLS related-code here ....
`mongodb/mongos/config/init.sls`       # <---  some duplicated code (i.e file.directory, users)
`mongodb/mongod/`
`mongodb/bic/config/init.sls`          #<-- some duplicated code (i.e. file.directory, users)
`mongodb/robo3t/`
`mongodb/compass/`

(2B) The opensds-formula has variation of this pattern.

`opensds/config/init.sls`     #<----- centralized duplicated SLS related-code here ....
`opensds/backend/config/init.sls`       # <---  some duplicated code (i.e file.directory)
`opensds/dashboard/`
`opensds/dock/config/init.sls`          #<-- some duplicated code (i.e. file.directory)
`opensds/infra/`
`opensds/hotpot/`
… etc …

**(2C) and one component with sub-component(s):**

`opensds/config/init.sls`     
`opensds/backend/config/init.sls`
`opensds/backend/block/config/init.sls`


**Example 3: Introducing Sub-Components.**

MongoDB has multiple GUI (robo3T, studio3T, compass) so following is possible.
None of the GUI sub-components currently need config.

`mongodb/mongos/config/init.sls`
`mongodb/mongod/`
`mongodb/bic/config/init.sls`
`mongodb/gui/robo3t/`                             #<  GUI
`mongodb/gui/compass/`                         #< GUI

**Example 4: Sub-component with config:**

If we extend component with a new sub-component requiring configuration....
But still there is no top-level config shared between component and sub-components.

`mongodb/mongos/config/init.sls`
`mongodb/mongod/`
`mongodb/bic/config/init.sls`
`mongodb/gui/robo3t/`                             
`mongodb/gui/compass/`                        
`mongodb/gui/studio3T/config/init.sls`   #< GUI

**Comment by myii** [comment]
@aboe76 Since @noelmcloughlin doesn't have any formulas where this can be implemented, shall we use this in `systemd-formula`, to confirm that we're happy with this functionality?

**Comment by aboe76** [comment]
@myii I'm ok with that.

**Comment by myii** [comment]
@baby-gnu Based on the comments above, would you mind using this PR as a basis for updating `systemd-formula`?  Once we get that working, we can confirm merging this PR here, so that it can be introduced in the other formulas as well.

**Comment by baby-gnu** [comment]
@myii ok, I'll do that.

**Comment by baby-gnu** [comment]
I made a pull request saltstack-formulas/systemd-formula#37 for review.

**Comment by baby-gnu** [comment]
I'll rebase this pull request as soon as we agree on saltstack-formulas/systemd-formula#37.

**Comment by baby-gnu** [comment]
> I'll rebase this pull request as soon as we agree on [saltstack-formulas/systemd-formula#37](https://github.com/saltstack-formulas/systemd-formula/pull/37).

Since it's merged in `systemd-formula`, I rebased this PR.

**Comment by myii** [comment]
@daks @noelmcloughlin @aboe76 Any concerns if we go ahead and merge this?  I'm OK with this after seeing how it has worked out in the `systemd-formula`.  I ask because once this is merged, it will be propagated to all of the other TOFS formulas, so I'd like to confirm that with you before that happens.

**Comment by aboe76** [comment]
@myii I have no problems with merging this.

**Comment by myii** [comment]
@baby-gnu Merged this.  Thanks for that confirmation, @aboe76.
